spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/feature/filters/TargetingFilter.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager.feature.filters;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.spring.cloud.feature.manager.FeatureFilter;
+import com.microsoft.azure.spring.cloud.feature.manager.TargetingException;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.Audience;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.GroupRollout;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.ITargetingContextAccessor;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingContext;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingEvaluationOptions;
+import com.microsoft.azure.spring.cloud.feature.manager.targeting.TargetingFilterSettings;
+
+public class TargetingFilter implements FeatureFilter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TargetingFilter.class);
+    
+    private static final String USERS = "users";
+    private static final String GROUPS = "groups";
+    private static final String AUDIENCE = "Audience";
+    
+    private static final String OUT_OF_RANGE = "The value is out of the accepted range.";
+    private static final String REQUIRED_PARAMETER = "Value cannot be null.";
+
+    private ITargetingContextAccessor contextAccessor;
+
+    private TargetingEvaluationOptions options;
```

Comment:
It's good to be final to avoid any accidental change in the class life cycle.
```suggestion
    private final TargetingEvaluationOptions options;
```